### PR TITLE
Explore restaurants

### PR DESCRIPTION
**Pull request by olaokoroafor**
Description:
 - I was able to set up structure of refreshing
 - Works sometimes, but the times it does not work do not seem to have a lot to do with the code
     - it seems like it is something to do with threading

### CODE REVIEW

File: FoodFriends/app/src/main/res/layout/item_restaurant.xml [hunk 1]
```diff
@@ -4,7 +4,8 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
-    android:layout_marginTop="60dp"
+    android:layout_marginTop="0dp"
+    android:layout_marginBottom="15dp"
```
Comment (nammyfb) [inline FoodFriends/app/src/main/res/layout/item_restaurant.xml:8]: Just a comment about having these in a dimens value file, so there's a consistent padding/margin throughout the app. Material theme generally have these as theirs. https://gist.github.com/aeroechelon/f6c205e04b241b43c80b

File: FoodFriends/app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -30,14 +30,11 @@
 
     <androidx.fragment.app.FragmentContainerView
         android:id="@+id/tvPlaceholder"
-        android:layout_width="419dp"
-        android:layout_height="611dp"
-        android:layout_marginTop="60dp"
+        android:layout_width="421dp"
```
Comment (nammyfb) [inline FoodFriends/app/src/main/res/layout/activity_main.xml:33]: Not sure how to comment on the whole file. But generally we'd want the main activity screen as simple as possible and control everything through fragments. ie: You'd have a single fragment container spanning width and height of the screen. 

For functionality like logout: I'd suggest having another tab like settings. Or even have a button in your profiles fragment that invokes the logout functionality

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ExploreFragment.java [hunk 3]
```diff
@@ -47,12 +48,14 @@
 public class ExploreFragment extends Fragment {
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ExploreFragment.java:48]: Now that we functionality for fetching and there's a mental model of how UI and business logic should flow together. We can probably look ahead and start using livedata objects to serve UI functionality. 

https://developer.android.com/topic/libraries/architecture/livedata

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ExploreFragment.java [hunk 4]
```diff
@@ -101,46 +113,38 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
-            }
 
-            @Override
-            public void onFailure(int statusCode, Header[] headers, String res, Throwable t) {
-                // called when response HTTP status is "4XX" (eg. 401, 403, 404)
-            }
-        });
-    }
-*/
     private void yelpQuery() {
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ExploreFragment.java:134]: This looks like a good candidate to be in a Model class

File: FoodFriends/app/src/main/java/com/example/foodfriends/models/Restaurant.java [hunk 5]
```diff
@@ -28,6 +31,7 @@ public class Restaurant extends ParseObject {
     public static final String LONGITUDE_KEY = "longitude";
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/models/Restaurant.java:31]: minor: Could these be private?

File: FoodFriends/app/src/main/java/com/example/foodfriends/models/Restaurant.java [hunk 6]
```diff
@@ -87,6 +91,67 @@ public void setAddress(String address){
...
+            r.setCity(location.getString("city"));
+            r.setState(location.getString("state"));
+            r.setZipcode(location.getString("zip_code"));
+            JSONObject coordinates = jsonObject.getJSONObject("coordinates");
+            r.setLatitude(coordinates.getDouble("latitude"));
+            r.setLongitude(coordinates.getDouble("longitude"));
+            r.setPrice("price");
+            String address = "";
+            JSONArray display =  location.getJSONArray("display_address");
+            for(int i = 0; i < display.length(); i++){
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/models/Restaurant.java:122]: minor: Code formatting looks a bit off. There's a nifty tool in android studio. Code -> Reformat Code,  and it'll handle all the formatting for you :)

**Review by nammyfb — CHANGES_REQUESTED** [review]

File: FoodFriends/app/src/main/java/com/example/foodfriends/models/Restaurant.java (hunk 6 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/models/Restaurant.java:122]: Thanks! Just reformatted this file, will probably also use this tool for other files

File: FoodFriends/app/src/main/res/layout/item_restaurant.xml (hunk 1 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/res/layout/item_restaurant.xml:8]: Have made a dimensions files, and now the text sizes and margin/padding sizes reference the file!

File: FoodFriends/app/src/main/res/layout/activity_main.xml (hunk 2 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/res/layout/activity_main.xml:33]: I have moved to log out functionality into the profile fragment and the activity main now has an empty fragment

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ExploreFragment.java (hunk 3 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ExploreFragment.java:48]: Implement Live Data Models

**Pull request merged**